subsys/random/rand32_ctr_drbg.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <init.h>
+#include <device.h>
+#include <entropy.h>
+#include <kernel.h>
+#include <string.h>
+
+#if defined(CONFIG_MBEDTLS)
+#if !defined(CONFIG_MBEDTLS_CFG_FILE)
+#include "mbedtls/config.h"
+#else
+#include CONFIG_MBEDTLS_CFG_FILE
+#endif /* CONFIG_MBEDTLS_CFG_FILE */
+#include <mbedtls/ctr_drbg.h>
+
+#elif defined(CONFIG_TINYCRYPT)
+
+#include <tinycrypt/ctr_prng.h>
+#include <tinycrypt/aes.h>
+#include <tinycrypt/constants.h>
+
+#endif /* CONFIG_MBEDTLS */
+
+static K_SEM_DEFINE(state_sem, 1, 1);
+
+static struct device *entropy_driver;
+
+#if defined(CONFIG_MBEDTLS)
+
+static mbedtls_ctr_drbg_context ctr_ctx;
+
+static int ctr_drbg_entropy_func(void *ctx, unsigned char *buf, size_t len)
+{
+	return entropy_get_entropy(ctx, (void *)buf, len);
+}
+
+#elif defined(CONFIG_TINYCRYPT)
+
+static TCCtrPrng_t ctr_ctx;
+
+#endif /* CONFIG_MBEDTLS */
+
+
+static int ctr_drbg_initialize(void)
+{
+	int ret;
+	static const unsigned char drbg_seed[] = "zephyr ctr-drbg seed";
```

Comment:
Maybe this should be a config item, since the idea is that a given application can set it to something unique to that application.